SU2_CFD/src/solvers/CNSSolver.cpp

```diff
@@ -535,7 +535,23 @@ void CNSSolver::BC_HeatFlux_Wall_Generic(const CGeometry *geometry, const CConfi
       }
 
       if (kind_boundary == HEAT_TRANSFER){
-        Jacobian.AddVal2Diag(iPoint, nDim+1, Transfer_Coefficient*Area);
+
+        /*--- It is necessary to zero the jacobian entries of the energy equation. ---*/
+        for (auto iVar = 0u; iVar < nVar; ++iVar)
+          Jacobian_i[nDim+1][iVar] = 0.0;
+
+        const su2double Density = nodes->GetDensity(iPoint);
+        const su2double Gas_Constant = config->GetGas_ConstantND();
+        const su2double Vel2 = GeometryToolbox::SquaredNorm(nDim, &nodes->GetPrimitive(iPoint)[1]);
+        const su2double dTdrho = 1.0/Density * ( -Tinfinity + (Gamma-1.0)/Gas_Constant*(Vel2/2.0) );
+
+        /*--- Equivalent to derivation in CNSSolver::BC_Isothermal_Wall_Generic, but no idea why that derivation is correct. ---*/
+        Jacobian_i[nDim+1][0] = Transfer_Coefficient * dTdrho * Area;
```

Comment:
I did not really understand why  dT\d rho  is what is displayed there. If someone can give me a pointer that would be appreciated.

It accelerates the convergence by a bit, actually quite a respectable bit from my test with navier_stokes/cylinder